controller/system_backup_controller.go

```diff
@@ -845,6 +857,112 @@ func (c *SystemBackupController) createVolumeBackup(volume *longhorn.Volume, sys
 	return backup, nil
 }
 
+func (c *SystemBackupController) BackupBackingImage() (map[string]*longhorn.BackupBackingImage, error) {
+	backingImages, err := c.ds.ListBackingImagesRO()
+	if err != nil {
+		return nil, err
+	}
+
+	backingImageBackups := make(map[string]*longhorn.BackupBackingImage, len(backingImages))
+	for _, backingImage := range backingImages {
+		backupBackingImage, err := c.createBackingImageBackup(backingImage)
+		if err != nil {
+			return nil, err
+		}
+
+		backingImageBackups[backupBackingImage.Name] = backupBackingImage
+	}
+
+	return backingImageBackups, nil
+}
+
+func (c *SystemBackupController) WaitForBackingImageBackupToComplete(backupBackingImages map[string]*longhorn.BackupBackingImage, systemBackup *longhorn.SystemBackup) {
+	var err error
+	log := getLoggerForSystemBackup(c.logger, systemBackup)
+
+	existingSystemBackup := systemBackup.DeepCopy()
+	defer func() {
+		record := &systemBackupRecord{}
+		if err != nil {
+			c.updateSystemBackupRecord(record,
+				systemBackupRecordTypeError, longhorn.SystemBackupStateError,
+				constant.EventReasonStart, err.Error(),
+			)
+		} else {
+			c.updateSystemBackupRecord(record,
+				systemBackupRecordTypeNormal, longhorn.SystemBackupStateGenerating,
+				constant.EventReasonStart, SystemBackupMsgStarting,
+			)
+
+		}
+
+		c.handleStatusUpdate(record, systemBackup, existingSystemBackup, err, log)
+	}()
+
+	startTime := time.Now()
+	ticker := time.NewTicker(time.Second * 5)
+	defer ticker.Stop()
+
+	for range ticker.C {
+		for name := range backupBackingImages {
+			// Retrieve the latest backup
+			var backupBackingImage *longhorn.BackupBackingImage
+			backupBackingImage, err = c.ds.GetBackupBackingImageRO(name)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					continue
```

Comment:
Should we do `delete(backupBackingImages, name)` if not found?

Reply:
We don't delete the the backup name `WaitForVolumeBackupToComplete` 
I think we can follow the same behavior here

The result of `not deleting the name when not found` is that it would definitely timeout in the end
because `if len(backupBackingImages) == 0` will never be zero

@c3y1huang 
WDYT?

Reply:
Is it possible that it is not found for a while, for example like the situation of etcd not synced?

Reply:
I think we can add a log message for the not found case
So user can debug more easily.

Reply:
> I think we can add a log message for the not found case So user can debug more easily.

Sounds good